ignite/base/events_driven.py

```diff
@@ -307,14 +316,15 @@ def __setattr__(self, attr: str, value: Any) -> None:
 
         super().__setattr__(attr, value)
 
-    def update_mapping(self, event_to_attr: Mapping[Any, str]) -> None:
+    def update_mapping(self, event_to_attr: Optional[Dict[Any, Any]]) -> None:
```

Comment:
Why it should be optional ? Empty method does not make much sense

Reply:
Mypy is complaining about it should be optional, because in `register_events` in Engine `event_to_attr` is optional https://github.com/pytorch/ignite/blob/b2ff869c07fae30640cd37416596b20e76fa4dd1/ignite/engine/engine.py#L228
so either we do this or we ignore mypy in the line I sent above

Reply:
or either you do `if event_to_attr is not None`